optuna/trial/_frozen.py

```diff
@@ -166,22 +166,6 @@ def __init__(
         self._distributions = distributions
         self._trial_id = trial_id
 
-    # Ordered list of fields required for `__repr__`, `__hash__` and dataframe creation.
-    # TODO(hvy): Remove this list in Python 3.7 as the order of `self.__dict__` is preserved.
-    _ordered_fields = [
-        "number",
-        "_values",
-        "datetime_start",
-        "datetime_complete",
-        "params",
-        "_distributions",
-        "user_attrs",
-        "system_attrs",
-        "intermediate_values",
-        "_trial_id",
-        "state",
-    ]
```

Comment:
Question: The keys of `self.__dict__` will be as follows:

```python
['_number', 'state', '_values', '_datetime_start', 'datetime_complete', '_params', '_user_attrs', '_system_attrs', 'intermediate_values', '_distributions', '_trial_id']
```

We can see some differences:
- Some field including `number`, `params`, `user_attrs` and `system_attrs` will be prefixed with `_`.
- The order of fields is different. For example, `_values` is the second item of `_ordered_field`, but it is the third one of `self.__dict__`.

But, I don't think they affect users because of the following two reasons:
- `hash` value will be changed, but `hash` values are different process by process originally. So, we cannot compare hash values of different processes even if we use the current master.
- I confirmed that `trial_str = repr(trial)` can be restored by `eval(trial_str)`.
- I also confirmed that dumped trial with `repr` in `master` can be restored by `eval` in this PR.

What do you think of it?

Reply:
Thanks for the insightful review. Actually, the items and order of `self.__dict__` and `_ordered_field` are different each other. But, as you confirmed, 

> I confirmed that trial_str = repr(trial) can be restored by eval(trial_str).
> I also confirmed that dumped trial with repr in master can be restored by eval in this PR.

The impact of this change on users would be minimal. So, we can accept it.

Reply:
Thank you for your reply. That makes sense.